### DISCUSSION TITLE
Get All Settings regardless of what groups

### DISCUSSION THREAD

**Discussion by valluminarias** [spatie/laravel-settings#24] [Q&A]
Hello.

Is it possible to get all the settings regardless of what group they belong?

Example, I have these `Settings`:
`GeneralSettings`,
`CompanySetting`,
`OtherSettings`

How can I get all these settings in just a single call ang get a single array of these settings?

**Comment by rubenvanassche** [comment]
Hi @valluminarias,

Sadly enough this isn't possible at the moment. Though I think this is a feature we want in the future. The problem is that I not yet have a good idea on how to create an API for such a thing.

**Comment by SvetoslavStefanov** [comment]
You could access available setting classes through the settings config file.

`  
public function index() {
    $result = [];
    $availableSettings = config('settings.settings');

    foreach ($availableSettings as $settingClass) {
      $settings = app($settingClass);

      $result[$settings->group()] = $settings->toArray();
    }
    
    return response()->json($result);
  }
`
